packages/calcite-components/src/components/tab-nav/tab-nav.tsx

```diff
@@ -406,4 +511,46 @@ export class TabNav {
       tabTitles[this.selectedTabId].focus();
     });
   }
+
+  private getOverflowIcons(): (VNode | VNode[]) | null {
+    const dir = getElementDir(this.el);
+    const { messages } = this;
+    const tabNavWidth = this.el.offsetWidth;
+    const tabTitles = Array.from(this.el.querySelectorAll("calcite-tab-title"));
+    const firstTitle = tabTitles[0].getBoundingClientRect();
+    const lastTitle = tabTitles[tabTitles.length - 1].getBoundingClientRect();
+
+    const isOverflowingEnd =
+      dir !== "rtl" ? lastTitle.right > tabNavWidth : lastTitle.right < tabNavWidth;
+    const isOverflowingStart = firstTitle.left < 0;
+
+    const getActionChevronDirection = (overflowDirection: string): VNode => {
+      const isEnd = overflowDirection === "end";
+      const dirActionClass: string = isEnd ? CSS.arrowEnd : CSS.arrowStart;
+      const dirChevronIcon: string = isEnd && dir !== "rtl" ? ICON.chevronRight : ICON.chevronLeft;
+      const dirText: string = isEnd ? messages.previousTabTitles : messages.nextTabTitles;
+
+      const dirScroll = () =>
+        isEnd ? this.scrollToNextTabTitles() : this.scrollToPreviousTabTitles();
+
+      return (
+        <calcite-action
+          class={dirActionClass}
+          icon={dirChevronIcon}
+          onClick={() => dirScroll()}
+          scale={this.scale}
+          text={dirText}
+        />
+      );
+    };
+
+    const showEndAction = getActionChevronDirection("end");
```

Comment:
Both the end and start action vars are being created, but only one used. Can you update them to be created when needed?

Reply:
Not sure if it was used before, but it is now.